Feat/referral histories

erwin-huang opened the pull request.

LukeSamkharadze commented on src/pages/Referral/Segments/LeadHistory.tsx:
```diff
@@ -46,12 +46,12 @@ const LeadHistory: React.FC<LeadHistoryProps> = ({
 
     const sumAmount = paginatedRewards[selectedAddress]
-      .map((record) => record.outputTokenAmount)
+      .map((record) => Number(record.outputTokenAmount))
```
Are you sure js can handle numbers in wei?

erwin-huang replied:
the outputTokenAmount is not in Wei, it's in gwei (for Koda), and in eth (for Kapex)

LukeSamkharadze commented on package.json:
```diff
@@ -52,6 +52,7 @@
     "fs": "^0.0.1-security",
     "get-port": "^6.0.0",
     "graphql": "^14.5.6",
+    "graphql-request": "^4.2.0",
```
I think we already have another graphql client packages installed, can you take a look at `queries.js` and `client.js` files?

erwin-huang replied:
Ok, I will check it out

LukeSamkharadze commented on src/pages/Referral/Segments/LeadInfluencer.tsx:
```diff
@@ -143,50 +144,20 @@ const LeadInfluencer: React.FC<SegmentsProps> = ({
   const [subReward, setSubReward] = useState<PaginatedRewards>({})
-  const { account, library } = useWeb3React()
+  const { account } = useWeb3React()
+  const referralHistories = useReferralHistories(account ?? "", outputToken?.address ?? "")
```
I think you can have optional arguments in `useReferralHistories` and avoid using `?? ""` everywhere

LukeSamkharadze commented on src/constants/graphs.ts:
```diff
@@ -0,0 +1,2 @@
+export const REFERRAL_CLIENT = process.env.REACT_APP_LINK_GRAPH_NODE_REFERRAL ?? "";
+export const BLOCK_CLIENT = process.env.REACT_APP_LINK_GRAPH_NODE_BLOCK ?? "";
```
Let's be consistent and use `'${}'` syntax like we have in `constants/index`

LukeSamkharadze reviewed (commented):
Are fetching all data at the same time? I think it can quite, big so what do you think about using some pagination (not actual pages) but load 20 and then have load more then fetch another 20 and like that

erwin-huang commented:
> Are fetching all data at the same time? I think it can quite, big so what do you think about using some pagination (not actual pages) but load 20 and then have load more then fetch another 20 and like that

Sound great, I have updated the codes

LukeSamkharadze reviewed (commented):
I don't think we should display mainnet history for testnet, so let's just show text that swap history is not supported on testnet

LukeSamkharadze commented on src/pages/Referral/SwapList.tsx:
```diff
@@ -1,89 +1,32 @@
-        <CustomLightSpinner src="/images/blue-loader.svg" alt="loader" size="45px" />
-      )}
-      {(swapList.length === 0 && !isLoading) && (
+      {referralHistories.length}
```
.

erwin-huang commented:
> swap history is not supported on current network

I have updated this one as well

LukeSamkharadze reviewed (commented):
This incrimental data fetching is good, but what do you think about if we added `load more` button that will fetch next 20 swaps. This still can get pretty resource intenstive if we display all of the swaps at the same time, so adding `load more` button will fix that problem

And also let's swap `Rewarded Tokens` and `Referred swap history` places, I think it will be good, since they won't have to scroll that much to claim reward

erwin-huang commented:
> This incrimental data fetching is good, but what do you think about if we added `load more` button that will fetch next 20 swaps. This still can get pretty resource intenstive if we display all of the swaps at the same time, so adding `load more` button will fix that problem
> 
> And also let's swap `Rewarded Tokens` and `Referred swap history` places, I think it will be good, since they won't have to scroll that much to claim reward

It's a great idea, I have updated referral histories.

but we can't do that for lead influencer histories (it will show sub influencer histories)
because currently, we don't have data about the lead's sub influencer in the subgraph
so we need all the histories and filter them to get the lead's sub influencer list

PS: we don't have the `SubInfluencer event` when we set a new sub influencer, so we don't have the data on subgraph

LukeSamkharadze commented:
> > This incrimental data fetching is good, but what do you think about if we added `load more` button that will fetch next 20 swaps. This still can get pretty resource intenstive if we display all of the swaps at the same time, so adding `load more` button will fix that problem
> > And also let's swap `Rewarded Tokens` and `Referred swap history` places, I think it will be good, since they won't have to scroll that much to claim reward
> 
> It's a great idea, I have updated referral histories.
> 
> but we can't do that for lead influencer histories (it will show sub influencer histories) because currently, we don't have data about the lead's sub influencer in the subgraph so we need all the histories and filter them to get the lead's sub influencer list
> 
> PS: we don't have the `SubInfluencer event` when we set a new sub influencer, so we don't have the data on subgraph

I think we can index those too
